src/madness/world/thread.h

```diff
@@ -1377,20 +1356,6 @@ namespace madness {
         /// \return True if a task was run.
         static bool run_task() {
 #ifdef HAVE_INTEL_TBB
-
-            // Construct a the parent task
-            tbb::task& waiter = *new( tbb::task::allocate_root() ) tbb::empty_task;
-            waiter.set_ref_count(2);
-
-            // Create a dummy task that we send throught the queue.
-            tbb::task& dummy = *new( waiter.allocate_child() ) tbb::empty_task;
-            tbb::task::enqueue(dummy);
-
-            // Run tasks while we wait for the dummy task.
-            waiter.wait_for_all();
-
-            // destroy the waiter
-            tbb::task::destroy(waiter);
```

Comment:
Is it some sort of moonlighting? We can try to implement similar approach with `task_hadnle`; however, the overall idea seems fragile.

Reply:
no